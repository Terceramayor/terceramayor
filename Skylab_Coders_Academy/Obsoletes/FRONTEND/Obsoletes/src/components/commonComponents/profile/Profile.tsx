// eslint-disable-next-line no-use-before-define
import React, { useEffect } from 'react';
import { View, Text, ScrollView } from 'react-native';
import Dispatch, { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loadUserProfile } from '../../../redux/actions/actionsObsoletesUserProfile';
import NavBar from './../../commonComponents/navBar/NavBar';
import ProductProfile from './../productProfile/ProductProfile';
import { reduxStateInterface, mapStateToPropsProfileReturnInterface, ProfileProps } from '../../../utils/interfaces';
import profileStyles from './profileStyles';

function Profile ({ userprofileProducts, navigation, userLogIn, actions }:ProfileProps) {
  useEffect(() => {
    actions.loadUserProfile(userLogIn.username);
  }, [userLogIn]);

  const {
    componentContainer,
    scrollViewFormat,
    helloUserProfile
  } = profileStyles;
  const { productsArray } = userprofileProducts;
  if (userLogIn.logInStatus === false) {
    return (
    <View style={componentContainer}>
  <NavBar navigation={navigation}/>
    <Text style={helloUserProfile}>No user signed in!</Text>
    </View>
    );
  }

  return (

<View style={componentContainer}>
  <NavBar navigation={navigation}/>

  {(productsArray?.length < 1)
    ? (

<Text style={helloUserProfile}>You have not yet provided feedback of any product!</Text>

      )
    : (
      <>
      <Text style={helloUserProfile}> Hello {userLogIn.username}, here you have all the products you have rated</Text>

      <ScrollView style={scrollViewFormat}>
        {productsArray?.map((product) => (

          <ProductProfile key={product.productName} singleProduct={product} navigation={navigation}/>

        ))}
        </ScrollView>
        </>
      )}

</View>
  );
}

const mapStateToProps = (state:reduxStateInterface):mapStateToPropsProfileReturnInterface => ({
  userprofileProducts: state.userprofileProducts,
  userLogIn: state.userLogIn

});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  actions: bindActionCreators({ loadUserProfile }, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
