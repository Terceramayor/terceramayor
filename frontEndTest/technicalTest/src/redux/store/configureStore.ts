import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';

function configureStore() {
  return createStore(
    rootReducer,

    compose(applyMiddleware(reduxImmutableStateInvariant(), thunk))

  );
}

export default configureStore;
