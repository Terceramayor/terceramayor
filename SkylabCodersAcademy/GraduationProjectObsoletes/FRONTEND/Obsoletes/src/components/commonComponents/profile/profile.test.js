// eslint-disable-next-line no-use-before-define
import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Profile from './Profile';

jest.mock('./../../commonComponents/navBar/NavBar', () => 'MockedNavBarComponent');
jest.mock('./../productProfile/ProductProfile', () => 'ProductProfileComponent');

const mockStore = configureStore([thunk]);

const mockedProductArray = [{
  _id: '6047b993a3106f69e02d67c7',
  originId: '30493083045',
  productName: 'MiXiao Lite xl 45',
  thumbnailUrl: 'https://i.ibb.co/X3Kj5gb/i.png',
  brand: 'SumSag',
  category: 'smartPhone',
  obsoletion: 0.14,
  stats: [
    {
      _id: '604762aed7d1028120f70fe4',
      buyDate: '2017-01-04T00:00:00.000Z',
      broken: false,
      brokenDate: null,
      user: 'Pablo',
      reason: ''
    }

  ],
  __v: 0,
  updatedDate: '2019-12-01T00:00:00.000Z'
},
{
  _id: '6047b993a3106294e02d67c7',
  originId: '30493083045',
  productName: 'MiXiao Lite sl 45',
  thumbnailUrl: 'https://i.ibb.co/X3Kj5gb/i.png',
  brand: 'SumSag',
  category: 'smartPhone',
  obsoletion: 0.14,
  stats: [
    {
      _id: '604762aed7d1028120f70fe4',
      buyDate: '2017-01-04T00:00:00.000Z',
      broken: false,
      brokenDate: null,
      user: 'Pablo',
      reason: ''
    }

  ],
  __v: 0,
  updatedDate: '2019-12-01T00:00:00.000Z'
}];

describe('Given the Profile component', () => {
  describe('When it is rendered without any yet rated products with the user not being logged out', () => {
    test('Then the SnapShot should match the rendered tree', () => {
      const store = mockStore({
        userprofileProducts: { productsArray: [] },
        userLogIn: {
          logInStatus: false,
          username: 'Fake_Name'
        }

      });
      const navigation = { navigate: jest.fn() };
      const tree = render(<Provider store={store} ><Profile navigation={navigation} userprofileProducts={mockedProductArray}/></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
  describe('When it is rendered with an already rated products array for an specific user and the user which is logged in', () => {
    test('Then the SnapShot should match the rendered tree', () => {
      const store = mockStore({
        userprofileProducts: { productsArray: mockedProductArray },
        userLogIn: {
          logInStatus: true,
          username: 'Fake_Name'
        }
      });

      const navigation = { navigate: jest.fn() };
      const tree = render(<Provider store={store} ><Profile navigation={navigation}/></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
  describe('When it is rendered with an already rated products empty array', () => {
    test('Then the SnapShot should match the rendered tree', () => {
      const store = mockStore({
        userprofileProducts: { productsArray: [] },
        userLogIn: {
          logInStatus: true,
          username: 'Fake_Name'
        }
      });

      const navigation = { navigate: jest.fn() };
      const tree = render(<Provider store={store} ><Profile navigation={navigation}/></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
});
