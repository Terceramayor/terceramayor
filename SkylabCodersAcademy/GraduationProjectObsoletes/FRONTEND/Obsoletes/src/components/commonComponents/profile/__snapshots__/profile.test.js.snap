// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Profile component When it is rendered with an already rated products array for an specific user and the user which is logged in Then the SnapShot should match the rendered tree 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#23485D",
      "flex": 1,
    }
  }
>
  <MockedNavBarComponent
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "#152B38",
        "fontFamily": "MontserratRegular",
        "fontSize": 17,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
     Hello 
    Fake_Name
    , here you have all the products you have rated
  </Text>
  <RCTScrollView
    style={
      Object {
        "width": "95%",
      }
    }
  >
    <View>
      <ProductProfileComponent
        navigation={
          Object {
            "navigate": [MockFunction],
          }
        }
        singleProduct={
          Object {
            "__v": 0,
            "_id": "6047b993a3106f69e02d67c7",
            "brand": "SumSag",
            "category": "smartPhone",
            "obsoletion": 0.14,
            "originId": "30493083045",
            "productName": "MiXiao Lite xl 45",
            "stats": Array [
              Object {
                "_id": "604762aed7d1028120f70fe4",
                "broken": false,
                "brokenDate": null,
                "buyDate": "2017-01-04T00:00:00.000Z",
                "reason": "",
                "user": "Pablo",
              },
            ],
            "thumbnailUrl": "https://i.ibb.co/X3Kj5gb/i.png",
            "updatedDate": "2019-12-01T00:00:00.000Z",
          }
        }
      />
      <ProductProfileComponent
        navigation={
          Object {
            "navigate": [MockFunction],
          }
        }
        singleProduct={
          Object {
            "__v": 0,
            "_id": "6047b993a3106294e02d67c7",
            "brand": "SumSag",
            "category": "smartPhone",
            "obsoletion": 0.14,
            "originId": "30493083045",
            "productName": "MiXiao Lite sl 45",
            "stats": Array [
              Object {
                "_id": "604762aed7d1028120f70fe4",
                "broken": false,
                "brokenDate": null,
                "buyDate": "2017-01-04T00:00:00.000Z",
                "reason": "",
                "user": "Pablo",
              },
            ],
            "thumbnailUrl": "https://i.ibb.co/X3Kj5gb/i.png",
            "updatedDate": "2019-12-01T00:00:00.000Z",
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;

exports[`Given the Profile component When it is rendered with an already rated products empty array Then the SnapShot should match the rendered tree 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#23485D",
      "flex": 1,
    }
  }
>
  <MockedNavBarComponent
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "#152B38",
        "fontFamily": "MontserratRegular",
        "fontSize": 17,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    You have not yet provided feedback of any product!
  </Text>
</View>
`;

exports[`Given the Profile component When it is rendered without any yet rated products with the user not being logged out Then the SnapShot should match the rendered tree 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#23485D",
      "flex": 1,
    }
  }
>
  <MockedNavBarComponent
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "#152B38",
        "fontFamily": "MontserratRegular",
        "fontSize": 17,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    No user signed in!
  </Text>
</View>
`;
